### PR TITLE
Add support for queries using Sequel's virtual rows

### PR DESCRIPTION
**Pull request by rosic**
This allows to easily query with inequality operators over the index
columns, e.g. [where](http://sequel.jeremyevans.net/rdoc/classes/Sequel/Dataset.html#method-i-where) { check_in_date > tomorrow }

> Virtual Rows - http://sequel.jeremyevans.net/rdoc/files/doc/virtual_rows_rdoc.html

### CODE REVIEW

**Comment by rosic** [comment]
cc: @marionzualo

File: lib/shameless/model.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 require 'shameless/index'
 require 'shameless/cell'
+require 'securerandom'
```
Comment (exviva) [inline lib/shameless/model.rb]: This is already in master 😊

File: spec/shameless/model_spec.rb [hunk 2]
```diff
@@ -171,6 +171,28 @@
     end
   end
 
+  describe '#where' do
```
Comment (exviva) [inline spec/shameless/model_spec.rb]: Could we add something similar in `index_spec.rb` that queries an explicit index?

File: spec/shameless/index_spec.rb [hunk 3]
```diff
@@ -13,18 +13,38 @@ class MyModel
...
+
+    expect(results.size).to eq(2)
+    expect(results.first[:foo]).to eq('bar')
+  end
+
+  it 'allows querying with block filters' do
+    MyModel.put(primary_id: 1, my_id: 1, check_in_date: today)
+    MyModel.put(primary_id: 2, my_id: 1, check_in_date: tomorrow)
+
+    results = MyModel.foo_index.where(my_id: 1) { |o| o.check_in_date > today }
```
Comment (exviva) [inline spec/shameless/index_spec.rb]: So sequel supports both `instance_eval`'ing the block, and passing an object to it?

Comment (rosic) [inline spec/shameless/index_spec.rb]: Exactly. So depending if you need to use local method methods variables, you'd one of those.

**Pull request merged**